File: backend/app/lib/supabase.py
# from supabase import create_client, Client
# import os

# def create_supabase_client() -> Client:
#     return create_client(os.getenv("SUPABASE_URL"), os.getenv("SUPABASE_KEY"))

from supabase import create_client, Client
import os

def create_supabase_client() -> Client:
    return create_client(os.getenv("SUPABASE_URL"), os.getenv("SUPABASE_KEY"))

File: SIH-AI-Farmer/frontend/src/lib/supabase.ts
import { createClient } from "@supabase/supabase-js";

   const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
   const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;

   export const supabase = createClient(supabaseUrl, supabaseAnonKey);

   export async function getUserHistory(userId: string) {
     const { data, error } = await supabase
       .from("user_history")
       .select("*")
       .eq("user_id", userId)
       .order("created_at", { ascending: false });
     if (error) throw new Error(error.message);
     return data;
   }